### PR TITLE
fix(accounts-db): inc snapshot + status cache verification

### PR DESCRIPTION
**Pull request by 0xNineteen**
- some hashes were default values (snapshots usually pre-load them, but not in this case)
- zero-lamport accounts + incremental is a special case (ie, is hash(pubkey) instead of hash(account))

with two fixes - incremental snapshots now validate successfully

---
- also needed to fix the status cache validation

### CODE REVIEW

File: src/accountsdb/snapshots.zig [hunk 1]
```diff
@@ -705,10 +705,17 @@ pub const StatusCache = struct {
...
             }
         }
-        for (slot_history.oldest()..slot_history.newest()) |slot| {
-            if (!slots_seen.contains(slot)) {
-                return error.SlotNotFoundInStatusCache;
+
+        var slots_checked: u32 = 0;
+        var slot = slot_history.newest();
+        while (slot >= slot_history.oldest() and slots_checked != MAX_CACHE_ENTRIES) {
+            if (slot_history.check(slot) == sysvars.SlotCheckResult.Found) {
```
[the remark this answers is not in the document]
Comment (0xNineteen) [reply src/accountsdb/snapshots.zig:712]: not always - more of a case-by-case basis where if its easier to read then you can include the full path - in this case, i agree with you

**Review by dnut — COMMENTED** [review]
looks good, bug seems fixed. just a couple minor comments

File: src/accountsdb/db.zig [hunk 2]
```diff
@@ -768,11 +769,28 @@ pub const AccountsDB = struct {
...
+                if (lamports == 0) {
+                    switch (config) {
+                        // for full snapshots, only include non-zero lamport accounts
+                        .FullAccountHash => continue,
+                        // zero-lamport accounts for incrementals = hash(pubkey)
+                        .IncrementalAccountHash => {
+                            var hasher = Blake3.init(.{});
+                            hasher.update(&key.data);
+                            hasher.final(&account_hash.data);
+                        },
```
[the remark this answers is not in the document]
Comment (0xNineteen) [reply src/accountsdb/db.zig:783]: 73e9ea6

File: src/accountsdb/snapshots.zig [hunk 3]
```diff
@@ -710,14 +710,21 @@ pub const StatusCache = struct {
...
             }
         }
-        for (slot_history.oldest()..slot_history.newest()) |slot| {
-            if (!slots_seen.contains(slot)) {
-                return error.SlotNotFoundInStatusCache;
+
+        var slots_checked: u32 = 0;
+        var slot = slot_history.newest();
+        while (slot >= slot_history.oldest() and slots_checked != MAX_CACHE_ENTRIES) {
+            if (slot_history.check(slot) == .Found) {
```
[the remark this answers is not in the document]
Comment (0xNineteen) [reply src/accountsdb/snapshots.zig:721]: 3564b91

**Review by InKryption — APPROVED** [review]

**Pull request merged**